### ISSUE TITLE
t\can_write_dir.t fails on Windows possibly because of incorrect OS check

### ISSUE THREAD

**Issue by nanis** [Perl-Toolchain-Gang/ExtUtils-Install#13]
I have been scratching my head over seemingly inexplicable test failures for a while now, and I think I just spotted the reason:

```text
> prove -b t\can_write_dir.t
t\can_write_dir.t .. 1/?
#   Failed test at t\can_write_dir.t line 50.
#     Structures begin differing at:
#          $got->[0] = '1'
#     $expected->[0] = '0'

#   Failed test at t\can_write_dir.t line 51.
#     Structures begin differing at:
#          $got->[0] = '1'
#     $expected->[0] = '0'
# Looks like you failed 2 tests of 12.
t\can_write_dir.t .. Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/12 subtests
```

The problem is not in the logic of `_can_write_dir`.  Nor is it in `_have_write_access`. Neither is the problem in Windows changing permissions under the script's feet.

`_have_write_access` depends on the flag `Is_Win32` to decide whether to use plain `-w` or `POSIX::access`:

```perl
            $has_posix = (!Is_cygwin && !Is_Win32
             && eval { local $^W; require POSIX; 1} ) || 0;
```

Running under my Visual Studio 2013 built `perl`, `!Is_cygwin` is `true`, and `eval { local $^W; require POSIX; 1}` returns `1` because indeed the `POSIX` module loads fine. So, the problem can only be with `Is_Win32`:

```perl
BEGIN {
    *Is_VMS        = $^O eq 'VMS'     ? sub(){1} : sub(){0};
    *Is_Win32      = $^O eq 'Win32'   ? sub(){1} : sub(){0};
    *Is_cygwin     = $^O eq 'cygwin'  ? sub(){1} : sub(){0};
    *CanMoveAtBoot = ($^O eq 'MSWin32' || $^O eq 'cygwin') ? sub(){1} : sub(){0};
}
```

Notice the simple typo? You are testing `$^O` against two distinct strings `'Win32'` and `'MSWin32'`. Only one comparison can be correct.

And, of course, as the platforms section in `perldoc perlport` confirms, `MSWin32` is the correct string. Fixing that typo fixes the tests.

I understand you may not want to depend on [Devel::CheckOS][1] to keep the number of dependencies small, but that would have avoided this problem.

This means the code has been subtly wrong on Windows everywhere `Is_Win32` is used.

[1]: https://metacpan.org/pod/Devel::CheckOS

**Comment by haarg** [comment]
Fixed in a8d181888d290a9405ae197a511a586872b6f0c3.

This isn't really a case of "want less dependencies" but "can't have external dependencies".

**Comment by nanis** [comment]
OK, makes sense. You are most welcome.
